src/router/handlers/utils.ts

```diff
@@ -5,5 +5,31 @@
  */
 
 import { createHash } from 'crypto';
+import * as xss from 'xss';
 
-export const hash = (o: any): string => createHash('sha256').update(JSON.stringify(o)).digest('hex');
+const options = {
+    whiteList: {
+        ...xss.whiteList,
+        div: ['xmlns'],
+    },
+    stripIgnoreTag: true,
+    stripIgnoreTagBody: ['script'],
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, consistent-return
+    onTagAttr(tag: string, name: string, value: string, isWhiteAttr: boolean): string | void {
+        if (name === 'xmlns') {
+            return `${name}=${value}`;
+        }
+    },
+};
+const xssValidator = new xss.FilterXSS(options);
+
+export const hash = (o: any): any => createHash('sha256').update(JSON.stringify(o)).digest('hex');
+
+export const validateXHTMLResource = (resource: any): any => {
+    if (process.env.VALIDATE_XHTML === 'true') {
```

Comment:
Does the validator strip off things that does not pass validate or does it fail request that does not pass validate? If we're enforcing HTML validate, we should just fail request that contains HTML. If we save the stripped version, it'll give requester the false impression that the request went through successfully and all contents were saved.

Reply:
This package only strips or escapes invalid requests, I couldn't find a clean way to throw an Error instead when values were stripped without inefficient string comparisons. However: in the case where a `patient.name[0].family` is set to `'<script>alert(123);</script>'`, the request will fail because the validation will leave the field empty, which fails the validator check. However, `'<script>alert(123);</script>test'` would leave the field as `test` and be accepted.

Reply:
Got it. Sounds like choosing another library that throws error is the most efficient way to go. Would comparing originalResource with validatedResource as strings work?